ObjC/Encoder+ObjC.mm

```diff
@@ -59,113 +59,101 @@ bool FLEncoder_WriteNSObject(FLEncoder encoder, id obj) FLAPI {
 }
 
 
+
 @implementation NSObject (Fleece)
 - (void) fl_encodeToFLEncoder: (FLEncoder)enc {
-    // Fall back to the internal fl_encodeTo:, which takes a raw C++ Encoder*.
-    [self fl_encodeTo: ((FLEncoderImpl*)enc)->fleeceEncoder.get()];
+    [self fl_encodeToFLEncoder: enc];
```

Comment:
This is the same method name as declared right below, and also it calls itself so it would crash ... just remove this `@implementation`.